source/docs/advanced/theme/route-dispatcher.md

```diff
@@ -0,0 +1,134 @@
+```js
+const resolvers = {
+  Query: {
+    matchUrls: (_, { url }, { loaders }) => {
+      return loaders.MyModuleUrl.matchBy(url)
+        .then((result) => {
+          if (result) {
+            return result;
+          } else {
+            return loaders.Url.matchBy(url);
+          }
+        })
+    },
+  }
+};
+```
+    Note that we don't want to forget to call the initial `loaders.Url.matchBy` here. If we don't, we will break the base functionalities of Front-Commerce.
+    
+    Additionally, since we are using an existing loader, we must make sure that our dependencies are correctly configured in our module. In case of a store using Magento, we should add the dependency on `Magento2/Url` module. See [Graphql module definition Reference](/docs/reference/graphql-module-definition.html#dependencies-optional) for more details.
+
+Once you've done these three steps, you should be able to test that everything works as expected by using the GraphQL Playground at http://localhost:4000/playground and executing the following query:
+
+```gql
+{
+  matchUrls(url: "my-dynamic-custom-url") {
+    url
+    type
+    identifier
+    target_path
+    redirect_type
+  }
+}
+```
+
+## Add the mapping between the `type` and the page component
+
+Once your server is correctly configured, you need to map the `type` that is returned in your `matchUrls` query to an actual component.
+
+To do so, you need to create the `my-module/web/moduleRoutes.js` file in your module that will contain the mapping. You might have already created if you followed the [Add a new page](/docs/essentials/add-a-page-client-side.html#Map-the-URL-to-the-page-component) guide. But instead of using the default export, you will need to export a named object `dispatchedRoutes`.
+
+This object has page types as keys (in our case `pageType`), and a render function as values that will tell the application what to render for a specific key (in our case, it renders `MyCustomPage`). This will give you something like this:
+
+```js
+//` my-module/web/moduleRoutes.js`
+import React from "react";
+import MyCustomPage from "theme/pages/MyCustomPage";
+
+export const dispatchedRoutes = {
+  pageType: props => <MyCustomPage id={props.matched.identifier} />
+};
+
+// you can still export your static routes here
+// export default () => [
+//   <Route ... />
+// ];
+```
+
+In the props passed to a render function (L6), you will have access to a `matched` property that is in fact the object returned by your `matchUrls` GraphQL query.
+
+Once you've created your file, you can refresh your application
+(`npm run start`), and you should see your new route if you go
+to the `/my-dynamic-custom-url` URL. It will display `MyCustomPage` component.
+
+And this is it! From now on, any URL that is matched to a particular `type` in your `matchUrls` query, will now be displayed with the render function you defined in your `dispatchedRoutes` export.
```

Comment:
```suggestion
And this is it! From now on, any URL that is matched to a particular `type` in your `matchUrls` query, will now be displayed with the render function defined in your `dispatchedRoutes` export under the matching key. Otherwise, a 404 page will still be displayed to the user.
```